code/game/objects/items/weapons/storage/dakis.dm

```diff
@@ -31,20 +32,20 @@
 			desc = "A large pillow depicting [custom_name] in a compromising position. Featuring as many dimensions as you."
 	else
 		if(!spam_flag)
-			spam_flag = 0
-			if(user.a_intent == "help")
+			spam_flag = FALSE
+			if("help")
 				user.visible_message("<span class='notice'>[user] hugs the [name].</span>")
-				playsound(src.loc, "rustle", 50, 1, -5)
-			if(user.a_intent == "disarm")
+				playsound(src, "rustle", 50, 1, -5)
+			if("disarm")
 				user.visible_message("<span class='notice'>[user] kisses the [name].</span>")
-				playsound(src.loc, "rustle", 50, 1, -5)
-			if(user.a_intent == "grab")
+				playsound(src, "rustle", 50, 1, -5)
+			if("grab")
 				user.visible_message("<span class='warning'>[user] holds the [name]!</span>")
-				playsound(src.loc, 'sound/items/bikehorn.ogg', 50, 1)
-			if(user.a_intent == "harm")
+				playsound(src, 'sound/items/bikehorn.ogg', 50, 1)
+			if("harm")
 				user.visible_message("<span class='danger'>[user] punches the [name]!</span>")
-				playsound(user.loc, 'sound/effects/shieldbash.ogg', 50, 1)
+				playsound(src, 'sound/effects/shieldbash.ogg', 50, 1)
 			spawn(cooldowntime)
```

Comment:
You don't actually need the spam flag at all, just change the click delay of the `user.changeNext_move(CLICK_CD_MELEE)`